### PR TITLE
log: various goodies

### LINKED ISSUE #1206: log: support job submission and log file rotation numbers
We currently allow viewing only the latest job submission, we should allow viewing previous ones too.

This will require an interface for querying the job submissions associated with a specific task on a specific flow (see [this issue](https://github.com/cylc/cylc-flow/issues/5247)).

We could present this as a drop-down option similar to the Cylc 7 GUI.

The Cylc log file "rotates" when the file gets too long, we could do with an interface for viewing previous rotations of this log file too.

Note the `cylc cat-log` command already provides support for both these features.

**Pull requests welcome!**

### LINKED ISSUE #1251: Log view usability improvements
- [x] Long workflow ID overflows input

  image

  If required the inputs should wrap onto the next line

- [x] Use a skeleton loader or indeterminate progress bar to indicate when log is loading. When viewing remote logs it can be quite a while before anything shows up.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### PR DESCRIPTION
**Pull request by oliver-sanders**
> **Built Atop:** https://github.com/cylc/cylc-ui/tree/log-tidy

> **Siblings:** https://github.com/cylc/cylc-uiserver/pull/439 & https://github.com/cylc/cylc-flow/pull/5453

> **Supersedes:** #1268 (PR got broke)

Closes #1206
Closes #1251

* Log files:
  * List available log files:
  * Support viewing other workflow log files e.g. `config/` and `install/`.
  * Support viewing logs from previous job submissions.
* Viewing:
  * Support stripping timestamps from logs.
  * Add a workflow/job toggle to switch between workflow and job logs.
  * Remove the search/update button by subscribing reactively.
  * Add a button for refreshing the log file list.
  * Disable the log file input when the file list is loading.
  * Display the path to the log file being viewed.
  * Display the connection status of the subscription and present a reconnect button to handle cat-log failures/timeouts.
* Opening:
  * Support clicking on jobs in the tree view to open the log for the appropriate job.
  * Make jobs click-able in the Graph view.
* Implementation:
  * Unify the `workflow` and `task` arguments into `id`.
  * Switch to using watchers and debounce for log file requests.
  * Use the ViewToolbar for presenting options.
  * Use a skeleton component for the loading state.

log-view

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [ ] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Review by hjoliver — APPROVED** [review]
Really good, the old GUI is really getting is ass kicked at this point. The code LGTM, had a good play with it, no problems seen.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 .DS_Store
 node_modules/
+node_modules
```
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply .gitignore:3]: I symlink my `node_modules` directory onto a faster (but less permanent) filesystem which makes it look like a file to git.

**Comment by wxtim** [comment]
Petty issue I've noticed - the refresh-file-list button changes the current file back to the default - can we avoid that?

**Review by wxtim — APPROVED** [review]
Some nice features.

**Review by wxtim — COMMENTED** [review]
Test failure locally 

  Subscription Component mixin
    ✓ should provide a hook for when the component is created
    1) should provide a hook for when the component is destroyed

**Review by wxtim — CHANGES_REQUESTED** [review]
Needs tests fixing

**Comment by oliver-sanders** [comment]
Fixed test & preserved the file selection for the "update log files" button.

**Review by wxtim — APPROVED** [review]
I've done a final check that it all looks functional locally 😄 

Still test failure?

**Pull request merged**